src/App.js

```diff
@@ -1,13 +1,21 @@
 import React from "react";
 import TabNav from "./components/TabNav.js";
 import Header from "./components/Header.js";
-
+import WelcomePage from "./components/WelcomePage.js";
+import CharacterList from './components/CharacterList';
+import LocationsList from './components/LocationsList';
+import EpisodesList from './components/EpisodeList';
+import { Route } from 'react-router-dom';
 
 export default function App() {
   return (
     <main>
       <Header />
-      {/* <TabNav /> */}
+      <TabNav />
+      <Route exact path='/' component={WelcomePage}/>
+      <Route exact path='/characters' component={CharacterList}/>
```

Comment:
while it works with the exact prop, if it's not needed it's best to leave it out